From 95776e3d38cf870e5c2a09d61dedc012013e878a Mon Sep 17 00:00:00 2001
From: Erick Narvaez <e-narvaez@ti.com>
Date: Thu, 10 Oct 2019 12:03:09 -0500
Subject: [PATCH] Virt: Fix init scripts to load virt drivers

Fixes a race condition seen in loading the kernel modules where modprobe
would load the pvrsrvkm driver before the init script could. This patch
will blacklist modprobe from doing so and use insmod for the init script
to ensure the correct driver is loaded.

Signed-off-by: Erick Narvaez <e-narvaez@ti.com>
---
 .../j721e_linux/wayland/release/etc/init.d/rc.pvr   | 13 ++-----------
 .../release/etc/modprobe.d/blacklist-pvr.conf       |  3 +++
 2 files changed, 5 insertions(+), 11 deletions(-)
 create mode 100644 targetfs/j721e_linux/wayland/release/etc/modprobe.d/blacklist-pvr.conf

diff --git a/targetfs/j721e_linux/wayland/release/etc/init.d/rc.pvr b/targetfs/j721e_linux/wayland/release/etc/init.d/rc.pvr
index 81c2d8e..5086111 100755
--- a/targetfs/j721e_linux/wayland/release/etc/init.d/rc.pvr
+++ b/targetfs/j721e_linux/wayland/release/etc/init.d/rc.pvr
@@ -48,16 +48,8 @@ load_pvr()
 
 	# Load the PVR Services module.
 	#
-	if ! /sbin/modprobe -q pvrsrvkm $PVR_SRVKM_PARAMS; then
-		echo "Module pvrsrvkm failed to load. Retrying."
-		if [ -z $depmod_has_been_run ]; then
-			if [ -e /sbin/depmod ]; then
-				echo "Running /sbin/depmod"
-				/sbin/depmod && depmod_has_been_run=1
-			fi
-		fi
-		if ! /sbin/modprobe -q pvrsrvkm $PVR_SRVKM_PARAMS ; then return; fi
-	fi
+	/sbin/insmod /lib/modules/$(uname -r)/extra/pvrsrvkm.ko $PVR_SRVKM_PARAMS
+	/usr/bin/pvrdebug -osonline 1,1
 
 	# Reload udev rules
 	#
@@ -65,7 +57,6 @@ load_pvr()
 		/sbin/udevadm control --reload-rules
 	fi
 
-
 	echo "Loaded PowerVR consumer services."
 	return 0;
 }
diff --git a/targetfs/j721e_linux/wayland/release/etc/modprobe.d/blacklist-pvr.conf b/targetfs/j721e_linux/wayland/release/etc/modprobe.d/blacklist-pvr.conf
new file mode 100644
index 0000000..ec9840c
--- /dev/null
+++ b/targetfs/j721e_linux/wayland/release/etc/modprobe.d/blacklist-pvr.conf
@@ -0,0 +1,3 @@
+# For loading virtualization driver
+install pvrsrvkm /bin/false
+install drm_nulldisp /bin/false
-- 
2.17.1

